mito-ai/src/components/AgentToolComponents/ErrorFixupToolUI.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import { OpenAI } from 'openai';
+import { GroupedErrorMessages } from '../../utils/chatHistory';
+import { classNames } from '../../utils/classNames';
+import { getContentStringFromMessage } from '../../utils/strings';
+import PythonCode from '../../Extensions/AiChat/ChatMessage/PythonCode';
+import AssistantCodeBlock from '../../Extensions/AiChat/ChatMessage/AssistantCodeBlock';
+import ExpandIcon from '../../icons/ExpandIcon';
+import AlertIcon from '../../icons/AlertIcon';
+import '../../../style/ErrorFixupToolUI.css';
+
+interface IErrorFixupToolUIProps {
+    messages: GroupedErrorMessages;
+    renderMimeRegistry: IRenderMimeRegistry;
+}
+
+const parsePythonErrorType = (content: string | undefined): string => {
+    if (!content) return 'Error';
+    const errorMatch = content.match(/(\w+Error):/);
+    return errorMatch?.[1] || 'Error';
+};
+
+const ErrorBlock = ({
+    errorMessage,
+    renderMimeRegistry,
+}: {
+    errorMessage: OpenAI.Chat.ChatCompletionMessageParam,
+    renderMimeRegistry: IRenderMimeRegistry,
+}) => {
+    const [expandedError, setExpandedError] = useState(false);
+
+    const errorContent = getContentStringFromMessage(errorMessage);
+    const errorType = parsePythonErrorType(errorContent);
+
+    const toggleError = (): void => {
+        setExpandedError(!expandedError);
+    };
+
+    return (
+        <div className={classNames('error-fixup-container', { expanded: expandedError })}>
+            <div
+                onClick={toggleError}
+                className={classNames('error-fixup-toggle', {
+                    expanded: expandedError
+                })}
+            >
+                <span className="error-fixup-toggle-content">
+                    <AlertIcon />
+                    {errorType} Detected
+                </span>
+                <ExpandIcon isExpanded={expandedError} />
+            </div>
```

Comment:
It would be nice to make this div into a component, called something like `AgentComponentHeader` or something like that. Because we have this same thing duplicated across a bunch of agent components now and we want them to be styled the same every time. 

I think it would need to be a component with properties: 
1. Icon
2. Text
3. OnClick
4. DisplayBorder

If making this a component isn't easy, then we can at least re-use the same classes for it.